src/sentry/plugins/migrator.py

```diff
@@ -25,37 +31,42 @@ def call(self):
                     # can disable the Plugin for all Projects. There'd be no
                     # Repos left, associated with the Plugin.
                     self.disable_for_all_projects(plugin)
+        self._logging_context()
 
-    def all_repos_migrated(self, provider):
+    def all_repos_migrated(self, provider: str) -> bool:
         return all(r.integration_id is not None for r in self.repos_for_provider(provider))
 
-    def disable_for_all_projects(self, plugin):
+    def disable_for_all_projects(self, plugin: Plugin2 | Plugin) -> None:
         for project in self.projects:
             try:
-                self.log(at="disable", project=project.slug, plugin=plugin.slug)
+                logger.info(
+                    "plugin.disabled", extra={"project": project.slug, "plugin": plugin.slug}
+                )
                 plugin.disable(project=project)
             except NotImplementedError:
                 pass
 
-    def repos_for_provider(self, provider):
+    def repos_for_provider(self, provider: str) -> list[RpcRepository]:
         return [r for r in self.repositories if r.provider == provider]
 
     @property
     def repositories(self) -> list[RpcRepository]:
         return repository_service.get_repositories(organization_id=self.organization.id)
 
     @cached_property
-    def projects(self):
+    def projects(self) -> list[RpcProject]:
         return list(self.organization.projects)
 
     @property
-    def plugins(self):
+    def plugins(self) -> list[Plugin2 | Plugin]:
         return [plugins.configurable_for_project(project) for project in self.projects]
 
-    @property
-    def _logging_context(self):
-        return {
-            "org": self.organization.slug,
-            "integration_id": self.integration.id,
-            "integration_provider": self.integration.provider,
-        }
+    def _logging_context(self) -> None:
+        logger.info(
+            "migrator-context",
+            extra={
+                "org": self.organization.slug,
+                "integration_id": self.integration.id,
+                "integration_provider": self.integration.provider,
+            },
+        )
```

Comment:
```suggestion
    def _logging_context(self, context: dict[str, Any]) -> dict[str, Any]:
        context.update({
            "org": self.organization.slug,
            "integration_id": self.integration.id,
            "integration_provider": self.integration.provider,
        })
        return context
```

To go with the suggestions above.